Deprecation warnings when building with NodeJS v14.15.1

After having resolved issues preventing rpi-ws281x from running with node.js 14.15.1, we now have some new deprecation warnings. I have chosen not resolve the issue as part of issue #9 as to not to increase the complexity of the ticket further.

The output with the deprecations:

```
make: Entering directory '/home/ajmas/development/third-party/rpi-ws281x-ajmas/build'
  CXX(target) Release/obj.target/rpi-ws281x/src/addon.o
../src/addon.cpp: In static member function ‘static Nan::NAN_METHOD_RETURN_TYPE Addon::render(Nan::NAN_METHOD_ARGS_TYPE)’:
../src/addon.cpp:201:49: warning: ‘v8::ArrayBuffer::Contents v8::ArrayBuffer::GetContents()’ is deprecated: Use GetBackingStore. See http://crbug.com/v8/9908. [-Wdeprecated-declarations]
     if ((uint32_t)(array->Buffer()->GetContents().ByteLength()) != (uint32_t)(4 * ws2811.channel[0].count))
                                                 ^
In file included from /home/ajmas/.cache/node-gyp/14.15.1/include/node/node.h:67,
                 from ../node_modules/nan/nan.h:56,
                 from ../src/addon.h:5,
                 from ../src/addon.cpp:1:
/home/ajmas/.cache/node-gyp/14.15.1/include/node/v8.h:5266:12: note: declared here
   Contents GetContents();
            ^~~~~~~~~~~
../src/addon.cpp:204:51: warning: ‘v8::ArrayBuffer::Contents v8::ArrayBuffer::GetContents()’ is deprecated: Use GetBackingStore. See http://crbug.com/v8/9908. [-Wdeprecated-declarations]
     if ((uint32_t)(mapping->Buffer()->GetContents().ByteLength()) != (uint32_t)(4 * ws2811.channel[0].count))
                                                   ^
In file included from /home/ajmas/.cache/node-gyp/14.15.1/include/node/node.h:67,
                 from ../node_modules/nan/nan.h:56,
                 from ../src/addon.h:5,
                 from ../src/addon.cpp:1:
/home/ajmas/.cache/node-gyp/14.15.1/include/node/v8.h:5266:12: note: declared here
   Contents GetContents();
            ^~~~~~~~~~~
../src/addon.cpp:207:65: warning: ‘v8::ArrayBuffer::Contents v8::ArrayBuffer::GetContents()’ is deprecated: Use GetBackingStore. See http://crbug.com/v8/9908. [-Wdeprecated-declarations]
     uint32_t *pixels = (uint32_t *)array->Buffer()->GetContents().Data();
                                                                 ^
In file included from /home/ajmas/.cache/node-gyp/14.15.1/include/node/node.h:67,
                 from ../node_modules/nan/nan.h:56,
                 from ../src/addon.h:5,
                 from ../src/addon.cpp:1:
/home/ajmas/.cache/node-gyp/14.15.1/include/node/v8.h:5266:12: note: declared here
   Contents GetContents();
            ^~~~~~~~~~~
../src/addon.cpp:208:64: warning: ‘v8::ArrayBuffer::Contents v8::ArrayBuffer::GetContents()’ is deprecated: Use GetBackingStore. See http://crbug.com/v8/9908. [-Wdeprecated-declarations]
     uint32_t *map = (uint32_t *)mapping->Buffer()->GetContents().Data();
                                                                ^
In file included from /home/ajmas/.cache/node-gyp/14.15.1/include/node/node.h:67,
                 from ../node_modules/nan/nan.h:56,
                 from ../src/addon.h:5,
                 from ../src/addon.cpp:1:
/home/ajmas/.cache/node-gyp/14.15.1/include/node/v8.h:5266:12: note: declared here
```

In making the changes, we should see whether we can support node.js 8 and if not decide whether requiring a more recent version of node.js is acceptable.

--- a/src/addon.cpp
+++ b/src/addon.cpp
@@ -199,15 +199,14 @@ NAN_METHOD(Addon::render)
     v8::Local<v8::Uint32Array> array = info[0].As<v8::Uint32Array>();
     v8::Local<v8::Uint32Array> mapping = info[1].As<v8::Uint32Array>();
 
-
-    if ((uint32_t)(array->Buffer()->GetContents().ByteLength()) != (uint32_t)(4 * ws2811.channel[0].count))
+    if ((uint32_t)(array->Buffer()->GetBackingStore()->ByteLength()) != (uint32_t)(4 * ws2811.channel[0].count))
         return Nan::ThrowError("Size of pixels does not match.");
 
-    if ((uint32_t)(mapping->Buffer()->GetContents().ByteLength()) != (uint32_t)(4 * ws2811.channel[0].count))
+    if ((uint32_t)(mapping->Buffer()->GetBackingStore()->ByteLength()) != (uint32_t)(4 * ws2811.channel[0].count))
         return Nan::ThrowError("Size of pixel mapping does not match.");
 
-    uint32_t *pixels = (uint32_t *)array->Buffer()->GetContents().Data();
-    uint32_t *map = (uint32_t *)mapping->Buffer()->GetContents().Data();
+    uint32_t *pixels = (uint32_t *)array->Buffer()->GetBackingStore()->Data();
+    uint32_t *map = (uint32_t *)mapping->Buffer()->GetBackingStore()->Data();
     uint32_t *leds = ws2811.channel[0].leds;
 
     for (int i = 0; i < ws2811.channel[0].count; i++) {

--- a/src/addon.cpp
+++ b/src/addon.cpp
@@ -2,13 +2,12 @@
 
 #include <sstream>
 
-#define DEFAULT_TARGET_FREQ     800000
-#define DEFAULT_GPIO_PIN        18
-#define DEFAULT_DMA             10
-#define DEFAULT_TYPE            WS2811_STRIP_RGB
+#define DEFAULT_TARGET_FREQ 800000
+#define DEFAULT_GPIO_PIN 18
+#define DEFAULT_DMA 10
+#define DEFAULT_TYPE WS2811_STRIP_RGB
 
 static ws2811_t ws2811;
-
 
 NAN_METHOD(Addon::configure)
 {
@@ -26,16 +25,15 @@ NAN_METHOD(Addon::configure)
         40, 41, 42, 43, 44, 45, 46, 46, 47, 48, 49, 50, 51, 52, 53, 54,
         55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70,
         71, 72, 73, 74, 76, 77, 78, 79, 80, 81, 83, 84, 85, 86, 88, 89,
-        90, 91, 93, 94, 95, 96, 98, 99,100,102,103,104,106,107,109,110,
-        111,113,114,116,117,119,120,121,123,124,126,128,129,131,132,134,
-        135,137,138,140,142,143,145,146,148,150,151,153,155,157,158,160,
-        162,163,165,167,169,170,172,174,176,178,179,181,183,185,187,189,
-        191,193,194,196,198,200,202,204,206,208,210,212,214,216,218,220,
-        222,224,227,229,231,233,235,237,239,241,244,246,248,250,252,255
-    };
+        90, 91, 93, 94, 95, 96, 98, 99, 100, 102, 103, 104, 106, 107, 109, 110,
+        111, 113, 114, 116, 117, 119, 120, 121, 123, 124, 126, 128, 129, 131, 132, 134,
+        135, 137, 138, 140, 142, 143, 145, 146, 148, 150, 151, 153, 155, 157, 158, 160,
+        162, 163, 165, 167, 169, 170, 172, 174, 176, 178, 179, 181, 183, 185, 187, 189,
+        191, 193, 194, 196, 198, 200, 202, 204, 206, 208, 210, 212, 214, 216, 218, 220,
+        222, 224, 227, 229, 231, 233, 235, 237, 239, 241, 244, 246, 248, 250, 252, 255};
 
-
-    if (ws2811.freq != 0) {
+    if (ws2811.freq != 0)
+    {
         return Nan::ThrowError("ws281x already configured.");
     }
 
@@ -55,8 +53,8 @@ NAN_METHOD(Addon::configure)
     ws2811.channel[1].brightness = 0;
     ws2811.channel[1].strip_type = 0;
 
-
-    if (info.Length() != 1 ) {
+    if (info.Length() != 1)
+    {
         return Nan::ThrowError("configure requires an argument.");
     }
 
@@ -68,7 +66,8 @@ NAN_METHOD(Addon::configure)
 
     ///////////////////////////////////////////////////////////////////////////
     // leds
-    if (Nan::Has(options, Nan::New<v8::String>("leds").ToLocalChecked()).ToChecked()) {
+    if (Nan::Has(options, Nan::New<v8::String>("leds").ToLocalChecked()).ToChecked())
+    {
         Nan::MaybeLocal<v8::Value> maybe_leds = Nan::Get(options, Nan::New<v8::String>("leds").ToLocalChecked());
         v8::Local<v8::Value> leds;
 
@@ -80,7 +79,8 @@ NAN_METHOD(Addon::configure)
 
     ///////////////////////////////////////////////////////////////////////////
     // dma
-    if (Nan::Has(options, Nan::New<v8::String>("dma").ToLocalChecked()).ToChecked()) {
+    if (Nan::Has(options, Nan::New<v8::String>("dma").ToLocalChecked()).ToChecked())
+    {
         Nan::MaybeLocal<v8::Value> maybe_dma = Nan::Get(options, Nan::New<v8::String>("dma").ToLocalChecked());
         v8::Local<v8::Value> dma;
 
@@ -90,18 +90,21 @@ NAN_METHOD(Addon::configure)
 
     ///////////////////////////////////////////////////////////////////////////
     // gpio
-    if (Nan::Has(options, Nan::New<v8::String>("gpio").ToLocalChecked()).ToChecked()) {
+    if (Nan::Has(options, Nan::New<v8::String>("gpio").ToLocalChecked()).ToChecked())
+    {
         Nan::MaybeLocal<v8::Value> maybe_gpio = Nan::Get(options, Nan::New<v8::String>("gpio").ToLocalChecked());
         v8::Local<v8::Value> gpio;
 
-        if (!maybe_gpio.IsEmpty() && maybe_gpio.ToLocal(&gpio)) {
+        if (!maybe_gpio.IsEmpty() && maybe_gpio.ToLocal(&gpio))
+        {
             ws2811.channel[0].gpionum = Nan::To<int>(gpio).FromMaybe(ws2811.channel[0].gpionum);
         }
     }
 
     ///////////////////////////////////////////////////////////////////////////
     // brightness
-    if (Nan::Has(options, Nan::New<v8::String>("brightness").ToLocalChecked()).ToChecked()) {
+    if (Nan::Has(options, Nan::New<v8::String>("brightness").ToLocalChecked()).ToChecked())
+    {
         Nan::MaybeLocal<v8::Value> maybe_brightness = Nan::Get(options, Nan::New<v8::String>("brightness").ToLocalChecked());
         v8::Local<v8::Value> brightness;
 
@@ -111,7 +114,8 @@ NAN_METHOD(Addon::configure)
 
     ///////////////////////////////////////////////////////////////////////////
     // stripType/strip/type
-    if (true) {
+    if (true)
+    {
         Nan::MaybeLocal<v8::Value> maybe_stripType;
         v8::Local<v8::Value> stripType;
 
@@ -124,39 +128,49 @@ NAN_METHOD(Addon::configure)
         if (Nan::Has(options, Nan::New<v8::String>("stripType").ToLocalChecked()).ToChecked())
             maybe_stripType = Nan::Get(options, Nan::New<v8::String>("stripType").ToLocalChecked());
 
-        if (maybe_stripType.ToLocal(&stripType)) {
+        if (maybe_stripType.ToLocal(&stripType))
+        {
             v8::String::Utf8Value value(v8::Isolate::GetCurrent(), Nan::To<v8::String>(stripType).ToLocalChecked());
             string stripTypeValue = string(*value);
 
-            if (stripTypeValue == "rgb") {
+            if (stripTypeValue == "rgb")
+            {
                 ws2811.channel[0].strip_type = WS2811_STRIP_RGB;
             }
-            else if (stripTypeValue == "rbg") {
+            else if (stripTypeValue == "rbg")
+            {
                 ws2811.channel[0].strip_type = WS2811_STRIP_RBG;
             }
-            else if (stripTypeValue == "grb") {
+            else if (stripTypeValue == "grb")
+            {
                 ws2811.channel[0].strip_type = WS2811_STRIP_GRB;
             }
-            else if (stripTypeValue == "gbr") {
+            else if (stripTypeValue == "gbr")
+            {
                 ws2811.channel[0].strip_type = WS2811_STRIP_GBR;
             }
-            else if (stripTypeValue == "brg") {
+            else if (stripTypeValue == "brg")
+            {
                 ws2811.channel[0].strip_type = WS2811_STRIP_BRG;
             }
-            else if (stripTypeValue == "bgr") {
+            else if (stripTypeValue == "bgr")
+            {
                 ws2811.channel[0].strip_type = WS2811_STRIP_BGR;
             }
-            else if (stripTypeValue == "grbw") {
+            else if (stripTypeValue == "grbw")
+            {
                 ws2811.channel[0].strip_type = SK6812_STRIP_GRBW;
             }
         }
-        else {
+        else
+        {
             ws2811.channel[0].strip_type = WS2811_STRIP_RGB;
         }
     }
     ws2811_return_t result = ws2811_init(&ws2811);
 
-    if (result) {
+    if (result)
+    {
         std::ostringstream errortext;
         errortext << "configure(): ws2811_init() failed: " << ws2811_get_return_t_str(result);
         return Nan::ThrowError(errortext.str().c_str());
@@ -165,12 +179,12 @@ NAN_METHOD(Addon::configure)
     info.GetReturnValue().Set(Nan::Undefined());
 };
 
-
 NAN_METHOD(Addon::reset)
 {
     Nan::HandleScope();
 
-    if (ws2811.freq != 0) {
+    if (ws2811.freq != 0)
+    {
         memset(ws2811.channel[0].leds, 0, sizeof(uint32_t) * ws2811.channel[0].count);
         ws2811_render(&ws2811);
         ws2811_fini(&ws2811);
@@ -179,14 +193,14 @@ NAN_METHOD(Addon::reset)
     ws2811.freq = 0;
 
     info.GetReturnValue().Set(Nan::Undefined());
-
 }
 
 NAN_METHOD(Addon::render)
 {
     Nan::HandleScope();
 
-    if (info.Length() != 2) {
+    if (info.Length() != 2)
+    {
         return Nan::ThrowError("render() requires pixels and pixel mapping arguments.");
     }
 
@@ -199,8 +213,8 @@ NAN_METHOD(Addon::render)
     v8::Local<v8::Uint32Array> array = info[0].As<v8::Uint32Array>();
     v8::Local<v8::Uint32Array> mapping = info[1].As<v8::Uint32Array>();
 
-    if ((uint32_t)(array->Buffer()->GetBackingStore()->ByteLength()) != (uint32_t)(4 * ws2811.channel[0].count))
-        return Nan::ThrowError("Size of pixels does not match.");
+    // if ((uint32_t)(array->Buffer()->GetBackingStore()->ByteLength()) != (uint32_t)(4 * ws2811.channel[0].count))
+    //     return Nan::ThrowError("Size of pixels does not match.");
 
     if ((uint32_t)(mapping->Buffer()->GetBackingStore()->ByteLength()) != (uint32_t)(4 * ws2811.channel[0].count))
         return Nan::ThrowError("Size of pixel mapping does not match.");
@@ -209,14 +223,14 @@ NAN_METHOD(Addon::render)
     uint32_t *map = (uint32_t *)mapping->Buffer()->GetBackingStore()->Data();
     uint32_t *leds = ws2811.channel[0].leds;
 
-    for (int i = 0; i < ws2811.channel[0].count; i++) {
+    for (int i = 0; i < ws2811.channel[0].count; i++)
+    {
         leds[i] = pixels[map[i]];
     }
 
     ws2811_render(&ws2811);
 
     info.GetReturnValue().Set(Nan::Undefined());
-
 };
 
 NAN_METHOD(Addon::sleep)
@@ -226,17 +240,14 @@ NAN_METHOD(Addon::sleep)
     usleep(info[0]->ToInt32(Nan::GetCurrentContext()).ToLocalChecked()->Value() * 1000);
 
     info.GetReturnValue().Set(Nan::Undefined());
-
 }
-
 
 NAN_MODULE_INIT(initAddon)
 {
-    Nan::SetMethod(target, "configure",  Addon::configure);
-    Nan::SetMethod(target, "render",     Addon::render);
-    Nan::SetMethod(target, "reset",      Addon::reset);
-    Nan::SetMethod(target, "sleep",      Addon::sleep);
+    Nan::SetMethod(target, "configure", Addon::configure);
+    Nan::SetMethod(target, "render", Addon::render);
+    Nan::SetMethod(target, "reset", Addon::reset);
+    Nan::SetMethod(target, "sleep", Addon::sleep);
 }
-
 
 NODE_MODULE(addon, initAddon);

